src/Thumbnail/LiipImagineThumbnail.php

```diff
@@ -78,13 +92,18 @@ public function generatePublicUrl(MediaProviderInterface $provider, MediaInterfa
      */
     public function generatePrivateUrl(MediaProviderInterface $provider, MediaInterface $media, $format)
     {
-        if (MediaProviderInterface::FORMAT_REFERENCE !== $format) {
-            throw new \RuntimeException('No private url for LiipImagineThumbnail');
-        }
-
         $path = $provider->getReferenceImage($media);
+        if (MediaProviderInterface::FORMAT_ADMIN === $format || MediaProviderInterface::FORMAT_REFERENCE === $format) {
+            return $path;
+        }
+        if (!$this->resolverRegistry instanceof ResolverRegistryInterface) {
+            throw new \RuntimeException(sprintf(
+                'Cannot generate private url for LiipImagine, use the "%s" added in 3.16 to add support.',
```

Comment:
This message will not be shown in previous versions.
```suggestion
                'Cannot generate private url for LiipImagine, use the "%s" to add support.',
```